tests/test_astrometryTask.py

```diff
@@ -99,18 +98,30 @@ def testUsedFlag(self):
                 count += 1
         self.assertEqual(count, len(results.matches))
 
-    def testMaxMeanDistance(self):
-        """If the astrometric fit does not satisfy the maxMeanDistanceArcsec
-        threshold, ensure task raises an lsst.pipe.base.TaskError.
+    def testWcsFailure(self):
+        """In the case of a failed WCS fit, test that the exposure's WCS is set
+           to None and the coord_ra & coord_dec columns are set to nan in the
+           source catalog.
         """
         self.exposure.setWcs(self.tanWcs)
         config = AstrometryTask.ConfigClass()
-        config.maxMeanDistanceArcsec = 0.0  # To ensure a "deemed" WCS failure
-        solver = AstrometryTask(config=config, refObjLoader=self.refObjLoader)
-        sourceCat = self.makeSourceCat(self.tanWcs, doScatterCentroids=True)
-
-        with self.assertRaisesRegex(pipeBase.TaskError, "Fatal astrometry failure detected"):
-            solver.run(sourceCat=sourceCat, exposure=self.exposure)
+        config.wcsFitter.order = 2
+        config.wcsFitter.maxScatterArcsec = 0.0  # To ensure a WCS failure
+        sourceSchema = afwTable.SourceTable.makeMinimalSchema()
+        measBase.SingleFrameMeasurementTask(schema=sourceSchema)  # expand the schema
+        # schema must be passed to the solver task constructor
+        solver = AstrometryTask(config=config, refObjLoader=self.refObjLoader, schema=sourceSchema)
+        sourceCat = self.makeSourceCat(self.tanWcs, sourceSchema=sourceSchema, doScatterCentroids=True)
+        results = solver.run(
+            sourceCat=sourceCat,
+            exposure=self.exposure,
+        )
+        # Check that matches is set to None, the sourceCat coord cols are all
+        # set to nan and that the WCS attached to the exposure is set to None.
+        self.assertTrue(results.matches is None)
+        self.assertTrue(np.all(np.isnan(sourceCat["coord_ra"])))
+        self.assertTrue(np.all(np.isnan(sourceCat["coord_dec"])))
+        self.assertTrue(self.exposure.getWcs() is None)
```

Comment:
This should also be testing the metadata (though maybe not to check if they're nans, per my previous comment.)

Reply:
I don't think I can check the metadata you're looking for here.  I did add checks for:
```
self.assertTrue(results.scatterOnSky is None)
self.assertTrue(results.matches is None)
```
and I have a placeholder check for the metadata in `ci_hsc_gen3` as:
```
    def testMetadataForFailedAstrom(self):
        """Test that the metadata for a failed astrometic fit is set properly.

        Wait until Eli weighs in on what he wants this to be before committing.
        """
        calexpMetadata = self.butler.get("calexp.metadata", self.calexpMinimalDataId)
        self.assertTrue(np.isnan(calexpMetadata["SFM_ASTROM_OFFSET_MEAN"]))
        self.assertTrue(np.isnan(calexpMetadata["SFM_ASTROM_OFFSET_STD"]))
```